src/datasets/mnist.nim

```diff
@@ -49,31 +49,33 @@
 # The labels values are 0 to 9.
 
 
-import  streams, endians, os,
-        ../tensor/tensor, ../io/io_stream_readers
+import  streams, endians, os, httpClient, strformat, future, sequtils,
+        ../tensor/tensor, ../io/io_stream_readers,
+        zip/gzipfiles
 
-proc read_mnist_images*(imgsPath: string): Tensor[uint8] {.noInit.}=
-  ## Load MNIST images into a Tensor[uint8]
+type mnist = tuple[
+  train_images: Tensor[uint8],
+  test_images: Tensor[uint8],
+  train_labels: Tensor[uint8],
+  test_labels: Tensor[uint8],
+]
+
+const tmp = getTempDir()
```

Comment:
I prefer "global" const in PascalCase to easily spot them.

Non-blocking: regarding datasets, they should probably be put in `.cache/arraymancer/` or something like that, not sure what PyTorch and Keras do. That can be changed later.

This will be important for pretrained models because they weight in hundreds of MB.